### PR TITLE
List conflicted branches

### LINKED ISSUE #1471: FR: Flag to only show conflicted branches
My main repository at work is a large-ish monorepo, so it has hundreds of branches at any given time. This is mostly fine, but when there is a conflict in some of them, filtering the list produced by `jj branch list` can be somewhat tedious (both because it is hard to find the specific branch, and because the conflict results in multiple lines of output that don't all contain the branch name). A workaround for this I've been using is `jj branch list | grep -A 6 conflicted` but it would be nice if jj could have a flag that only shows the conflicted branches and the conflict information.

### PR DESCRIPTION
**Pull request by Kintaro**
As requested in #1471, I added a new flag for `jj branch list` to only show branches that are conflicted.

Closes #1471

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by martinvonz — COMMENTED** [review]
Thanks! Could you also add a test case in `test_branch_command.rs` to show that it works and to prevent regressions?

**Comment by Kintaro** [comment]
Sure, I'll whip one up and update. Also seems like I need to address some build checks.

File: cli/src/commands/branch.rs [hunk 1]
```diff
@@ -672,10 +676,17 @@ fn cmd_branch_list(
     let mut formatter = ui.stdout_formatter();
     let formatter = formatter.as_mut();
 
-    let branches_to_list = view.branches().filter(|&(name, _)| {
+    let branches_to_list = view.branches().filter(|(name, target)| {
+        let has_conflict = target
+            .local_target
+            .as_normal()
+            .and_then(|id| repo.store().get_commit(id).ok())
+            .map_or(false, |commit| commit.has_conflict().unwrap_or(false));
```
Comment (yuja) [inline cli/src/commands/branch.rs:684]: `commit.has_conflict()` means the commit content (= trees) has conflict, but I guess "conflict" in #1471 is conflict of branch targets (or diverged branch?)

fyi, the former can be filtered by `-r 'conflict()'`.

Comment (Kintaro) [reply cli/src/commands/branch.rs:684]: Ah, I'll rewrite it then.

**Comment by Kintaro** [comment]
Accidentally closed the PR, had some issues with my fork.

**Pull request closed without merging**